kie-wb-common-react/boxed-expression-editor/src/components/LiteralExpression/LiteralExpression.tsx

```diff
@@ -51,16 +64,23 @@ export const LiteralExpression: React.FunctionComponent<LiteralExpressionProps>
     []
   );
 
-  return (
-    <div className="literal-expression">
+  const renderLiteralExpressionHeader = useCallback(
+    () => (
       <div className="literal-expression-header">
         <p className="expression-name">{expressionName}</p>
         <p className="expression-data-type">({expressionDataType})</p>
       </div>
+    ),
+    [expressionDataType, expressionName]
+  );
```

Comment:
Any particular reason for having a separate function generate the header? It's simple enough to be rendered inside the component return function

Reply:
Yes, I am calling that function only when `LiteralExpression` is not headless, i.e. `{!isHeadless ? renderLiteralExpressionHeader() : null}`

Reply:
Yes, I know. But to be honest, it looks less readable. AFAIK, there is no issue to have a separate function, especially with the usage of `useCallback`. If there is a metter of performance or robustness, I can change it for sure, please let me know

Reply:
Indeed, your solution is perfectly fine.
As per readability, it's subjective. I usually try to avoid fragmenting the code too much if it's not necessary.
This is not the case, but if you have a callback for every conditional rendering in longer templates, it could lead to have to jump up and down through code to have an idea of the whole component output.
That's my personal taste by the way. Feel free to stick with `renderLiteralExpressionHeader`.